STM32F405RGTx/Core/Src/rtos_threads/coulomb_counting_thread.cpp

```diff
@@ -4,51 +4,60 @@
 #include "bms_entry.hpp"
 #include "threads.hpp"
 
-#define MEASUREMENT_PERIODICITY_SECONDS MEASUREMENT_PERIODICITY * 1E-3
-#define NOMINAL_CAP     21600
-#define INIT_SOC        100
-
 RTOSThread CoulombCountingThread::thread;
 float CoulombCountingThread::prev_current;
+uint32_t CoulombCountingThread::prev_time;
+
+osThreadId_t CoulombCountingThread::getThreadId() { return CoulombCountingThread::thread.getThreadId(); }
 
 void CoulombCountingThread::initialize() {
     thread = RTOSThread(
         "coulomb_counting_thread",
         1024,
-        osPriorityNormal,
+        COULOMB_COUNTING_THREAD_PRIORITY,
         runCoulombCounting
     );
 }
 
 float CoulombCountingThread::getCharge() {
     // integrate wrt time to get charge
     float curr_current = global_bms_data.battery.current;
-    float trapArea = 0.5 * MEASUREMENT_PERIODICITY_SECONDS * (prev_current + curr_current);
-    prev_current = curr_current;
-    
+    uint32_t curr_time = HAL_GetTick();
+
+    float elapsed_time = (curr_time - CoulombCountingThread::prev_time)*1E-3;
+
+    float trapArea = 0.5*elapsed_time*(prev_current + curr_current);
+
+    CoulombCountingThread::prev_current = curr_current;
+    CoulombCountingThread::prev_time = curr_time;
+
     return trapArea;
 }
 
 void CoulombCountingThread::runCoulombCounting(void* args) {
     float totalChargeConsumed = 0.0;
+    prev_time = HAL_GetTick();
```

Comment:
Can you just move this to line 26 and instead of HAL_GetTick, use curr_time to replace prev time

Reply:
I'm not sure what you mean here sry. Also, what would be the advantage?

Reply:
oh nevermind im stupid this is outside the while loop